src/modules/track_model/src/track_model.cc

```diff
@@ -1 +1,339 @@
+{
+    return types::ERROR_NONE;
+}
+
+types::Error SoftwareTrackModel::SetYellowTrafficLight(const types::BlockId block, const bool on)
+{
+    return types::ERROR_NONE;
+}
+
+types::Error SoftwareTrackModel::SetGreenTrafficLight(const types::BlockId block, const bool on)
+{
+    return types::ERROR_NONE;
+}
+
+types::Error SoftwareTrackModel::SetCommandedSpeed(const types::BlockId block, const types::MetersPerSecond speed)
+{
+    if (blocks_.size() > block && block > 0)
+    {
+        for (int i = 0; i < trains_.size(); i++)
+        {
+            for (int j = 0; j < occupied_train_blocks_[i].size(); j++)
+            {
+                if (occupied_train_blocks_[i][j] == block)
+                {
+                    trains_[i]->SetCommandedSpeed(speed);
+                }
+            }
+        }
+
+        return types::ERROR_NONE;
+    }
+    else
+    {
+        return types::ERROR_INVALID_BLOCK;
+    }
+}
+
+types::Error SoftwareTrackModel::SetAuthority(const types::BlockId block, const types::Blocks authority)
+{
+    if (blocks_.size() > block && block > 0)
+    {
+        for (int i = 0; i < trains_.size(); i++)
+        {
+            for (int j = 0; j < occupied_train_blocks_[i].size(); j++)
+            {
+                if (occupied_train_blocks_[i][j] == block)
+                {
+                    trains_[i]->SetAuthority(authority);
+                }
+            }
+
+        }
+
+        return types::ERROR_NONE;
+    }
+    else
+    {
+        return types::ERROR_INVALID_BLOCK;
+    }
+}
+
+types::Error SoftwareTrackModel::GetBlockOccupancy(const types::BlockId block, bool &occupied) const
+{
+    //checking if block exists
+    if (blocks_.size() > block && block > 0)
+    {
+        occupied = blocks_[block].occupied;
+
+        return types::ERROR_NONE;
+    }
+    else
+    {
+        return types::ERROR_INVALID_BLOCK;
+    }
+}
+
+types::Error SoftwareTrackModel::SetBrokenRail(const types::BlockId block, const bool broken)
+{
+    return types::ERROR_NONE;
+}
+
+types::Error SoftwareTrackModel::SetTrackCircuitFailure(const types::BlockId block, const bool track_circuit_failure)
+{
+    return types::ERROR_NONE;
+}
+
+types::Error SoftwareTrackModel::SetPowerFailure(const types::BlockId block, const bool power_failure)
+{
+    return types::ERROR_NONE;
+}
+
+types::Error SoftwareTrackModel::SetExternalTemperature(const types::DegreesFahrenheit temperature)
+{
+    return types::ERROR_NONE;
+}
+
+//is this getting callled only when deboarding is gonna happen?
+types::Error SoftwareTrackModel::SetPassengersDeboarding(const types::TrainId train, const uint16_t passengers)
+{
+    if (trains_.size() > train)
+    {
+        const uint16_t deboarding = passengers;
+
+        //subtract from total passengers
+        uint16_t vacancy = train_capacity - passenger_counts_[train] + deboarding;
+
+        //generate random number within bounds for boarding
```

Comment:
This appears to be the same random number generator as in the Train Model.  Since this is being used in multiple places, move to `common` and update to match coding standard